WIP-extension.md

```diff
@@ -1,7 +1,33 @@
 Q:
 
-- [ ] Is FE aware of current crowdfunding phase?
+- [x] Is FE aware of current crowdfunding phase? _Yes_
 
+Flow to determine whether a prolong `packageOption` is returned.
+
+Input:
+- A membership
+- A prolong packageOption, to check against
+
+a)  Does user own membership, or pledge membership (ABO_GIVE)
+    - If false, end and don't return package option
+b)  Is membership active
+    - If false, end
+c)  Is current membershipPeriod last period
+    - If false, end
+d) Can membership.type be extend
+    - If false, end
+e)  If membership.membershipType != option.membershipType
+    - If true, indicate generation of new membership in customization
+    payload, then proceed
```

Comment:
This is for upgrades/downgrades right? If so option should have something like compatibleMembershipTypes and this check would be `if option.compatibleMembershipTypes.indexOf(membership.membershipType) > -1`

Reply:
It is, yes.

There is currently no scenario planned on a business in which you could not switch from one type to the other e.g. you can't switch from `BENEFACTOR` to `ABO_GIVE`-esque membership.

Reply:
> There is currently no scenario planned on a business in which you could not switch from one type to the other e.g. you can't switch from BENEFACTOR to ABO_GIVE-esque membership.

Can you elaborate? I read this two statements as oposing each other.
As you cannot switch from any membershipType to any other I would store in the membershipTypes table to which membershipTypes one can be upgraded. Or vice-versa: a membershipType sais from which membershipType can be upgraded to it.

Reply:
There is no (business) scenario planned in which you can switch from `BENEFACTOR` to `ABO_GIVE`. A benefactor membership would allow to extend with `ABO` or `BENEFACTOR`. But you can't turn your benefactor membership into a "giftable" membership.

`customPackages` won't provide you with a `ABO_GIVE` "package option".

You can extend a membership you pledged but gave away of `membershipType` `ABO` (there is no `ABO_GIVE` type). That transition is typoe `ABO` to `ABO`.

Am I making any sense? Am I missing here something?